Frameworks/CoreGraphics/CGDataConsumer.mm

```diff
@@ -16,56 +16,112 @@
 
 #import <StubReturn.h>
 #import <CoreGraphics/CGDataConsumer.h>
+#import <CFCppBase.h>
+#import <Starboard/SmartTypes.h>
+#import "CGDataConsumerInternal.h"
+
+#pragma region CGDataConsumer
+
+static const CGDataConsumerCallbacks sc_CGDataConsumerCFDataCallbacks{ [](void* info, const void* data, size_t size) {
+                                                                          CFDataAppendBytes(static_cast<CFMutableDataRef>(info),
+                                                                                            (const UInt8*)data,
+                                                                                            size);
+                                                                          return size;
+                                                                      },
+                                                                       [](void* info) { CFRelease(static_cast<CFTypeRef>(info)); } };
+
+static const CGDataConsumerCallbacks sc_CGDataConsumerCFURLCallbacks{
+    [](void* info, const void* data, size_t size) {
+        auto dataWrapper = woc::MakeAutoCF<CFDataRef>(CFDataCreateWithBytesNoCopy(nullptr, (const UInt8*)data, size, kCFAllocatorNull));
+        int errorCode = 0;
+        CFURLWriteDataAndPropertiesToResource(static_cast<CFURLRef>(info), dataWrapper, nullptr, &errorCode);
```

Comment:
Documentation is light as to what exactly it should be doing other than "When Core Graphics is ready to send data to the consumer, your function is called. It should copy the specified number of bytes from buffer into some resource under your control—for example, a file." I took this to mean writing to the resource is the expected behavior for URL.